SmartPlaces.Facilities/samples/Telemetry.Mapped/src/Processors/EventHubProcessor.cs

```diff
@@ -0,0 +1,336 @@
+﻿// -----------------------------------------------------------------------
+// <copyright file="EventHubProcessor.cs" company="Microsoft">
+// Copyright (c) Microsoft Corporation.  All rights reserved.
+// </copyright>
+// -----------------------------------------------------------------------
+
+namespace Telemetry.Processors
+{
+    using System;
+    using System.Collections.Generic;
+    using System.Globalization;
+    using System.Threading;
+    using System.Threading.Tasks;
+    using Azure;
+    using Azure.Core;
+    using Azure.Messaging.EventHubs;
+    using Azure.Messaging.EventHubs.Consumer;
+    using Azure.Messaging.EventHubs.Primitives;
+    using Azure.Storage.Blobs;
+    using Azure.Storage.Blobs.Models;
+    using Microsoft.Extensions.Logging;
+
+    /// <summary>
+    /// A generic handler for processing eventHub messages in a batch to enable
+    /// higher throughput reducing the overhead of checkpointing each message while
+    /// also enabling higher resource untilization of the available compute.
+    /// </summary>
+    /// <typeparam name="TPartition">A custom Partition</typeparam>
+    public class EventHubProcessor<TPartition> : EventProcessor<TPartition>
```

Comment:
I'm not sufficiently familiar with this API to provide a thorough review; will read up on it if required though, let me know.

Reply:
This code comes largely from https://devblogs.microsoft.com/azure-sdk/custom-event-processor/ and has been very slightly modified to fit our needs, particularly around OnProcessEventBatchAsync and the constructor accepting a Func to abstract the logic of how to process each twin out of this class.